kolibri/content/models.py

```diff
@@ -47,11 +47,11 @@ def get_db_prep_value(self, value, connection, prepared=False):
                 value = uuid.UUID(value)
             except AttributeError:
                 raise TypeError(self.error_messages['invalid'] % {'value': value})
-
-        if connection.features.has_native_uuid_field:
-            return value
         return value.hex
 
+    def from_db_value(self, value, expression, connection, context):
+        return self.to_python(value)
```

Comment:
It calls `to_python`, which is a couple lines down from here, which converts to hex if needed.

The other parameters are standard `from_db_value` args. I'm not taking advantage of them here.